website/docs/cli/commands/test/index.mdx

```diff
@@ -0,0 +1,368 @@
+| [`providers`](#the-providers-block)                                     | object            | Aliases for providers.                                                                                                                                                                                         |
+
+### The `run.assert` block
+
+You can specify `assert` blocks inside your `run` block to test the state of your infrastructure after the
+`apply` or `plan` operation is complete. There is no theoretical limit to the number of blocks you can define.
+
+Each block requires the following two attributes:
+
+1. The `condition` is an [OpenTofu condition](/docs/language/expressions/custom-conditions#condition-expressions) which
+   should return `true` for the test to pass, `false` for the test to fail. The condition **must** reference a
+   resource, data source, variable, output or module from the main code, otherwise OpenTofu will refuse to run the test.
+2. The `error_message` is a string explaining what happened when the test fails.
+
+:::tip Example
+
+As a simple example, you can write an `assert` block as follows:
+
+<Tabs>
+    <TabItem label={"main.tftest.hcl"} value={"test"} default>
+        <CodeBlock language="hcl">{SimpleTest}</CodeBlock>
+    </TabItem>
+    <TabItem label={"main.tf"} value={"main"}>
+        <CodeBlock language="hcl">{SimpleMain}</CodeBlock>
+    </TabItem>
+</Tabs>
+
+:::
+
+Please note that conditions only let you perform basic checks on the current OpenTofu state and use OpenTofu functions.
+**You cannot define additional data sources directly in your test code.** To work around this limitation, you can use
+[the `module` block](#the-runmodule-block) in order to load a helper module.
+
+### The `run.module` block
+
+In some cases you may find that the tools provided in the
+[condition expression](/docs/language/expressions/custom-conditions#condition-expressions) are not enough to test
+if your code created the infrastructure correctly.
+
+You can use the `module` block to override the main module `tofu test` loads. This gives you the opportunity to create
+additional resources or data sources that you can use in your `assert` conditions.
+
+Its syntax is similar to loading modules in normal OpenTofu code:
+
+```hcl
+run "test" {
+  module {
+    source = "./some-module"
+  }
+}
+```
+
+The `module` block has the following two attributes:
+
+* The `source` attribute points to the directory of the module to load or any other [module source](/docs/language/modules/sources).
+* The `version` specifies the version of the module you want to use.
+
+:::warning Note
+
+You cannot pass parameters directly in the `module` block as you may be used to from the normal OpenTofu code. Use the
+[`variables` block](#the-variables-and-runvariables-blocks) to pass parameters to your module.
+
+:::
+
+:::tip Example
```

Comment:
I'm not 100% behind the idea of using the `:::tip` block for this entire example. Maybe a heading would be nice then people can find it in the ToC?

Reply:
We can add a separate L3 heading, L4's don't show up in the ToC. It felt like this isn't important enough to be on the same level as the reference manual items. I'm not opposed to it either though.

Reply:
Heading would mean that it gets pulled out of the very large tip box (which feels a bit strange).

But I'm 50/50 on this one. Let's keep it as it is unless someone else feels strongly about it. 
For reference, this is how it renders:
![image](https://github.com/opentofu/opentofu/assets/237469/b9493419-e74e-4fa4-9a7e-3bcf5b8063e3)